colorcoder/Main.java

```diff
@@ -1,120 +1,35 @@
 package colorcoder;
 
 public class Main {
-    enum MajorColor {
-        WHITE(0),
-        RED(1),
-        BLACK(2),
-        YELLOW(3),
-        VIOLET(4);
-        private int index;
-        private MajorColor(int index) {
-            this.index = index;
-        }
-        int getIndex() {
-            return index;
-        }
-        public static MajorColor fromIndex(int index) {
-            for(MajorColor color: MajorColor.values()) {
-                if(color.getIndex() == index) {
-                    return color;
-                }
-            }
-            return null;
-        }
-    };
-    enum MinorColor {
-        BLUE(0),
-        ORANGE(1),
-        GREEN(2),
-        BROWN(3),
-        SLATE(4);
-        private int index;
-        private MinorColor(int index) {
-            this.index = index;
-        }
-        int getIndex() {
-            return index;
-        }
-        public static MinorColor fromIndex(int index) {
-            for(MinorColor color: MinorColor.values()) {
-                if(color.getIndex() == index) {
-                    return color;
-                }
-            }
-            return null;
-        }
-    };
 
-    final static String MajorColorNames[] = {
-        "White", "Red", "Black", "Yellow", "Violet"
-    };
-    final static int numberOfMajorColors = MajorColorNames.length;
-    final static String MinorColorNames[] = {
-        "Blue", "Orange", "Green", "Brown", "Slate"
-    };
-    final static int numberOfMinorColors = MinorColorNames.length;
+	static ColorPair getColorFromPairNumber(int pairNumber) {
+		int zeroBasedPairNumber = pairNumber - 1;
+		MajorColor majorColor = MajorColor.fromIndex(zeroBasedPairNumber / ColorPair.NUMBEROFMAJORCOLORS);
+		MinorColor minorColor = MinorColor.fromIndex(zeroBasedPairNumber % ColorPair.NUMBEROFMINORCOLORS);
+		return new ColorPair(majorColor, minorColor);
+	}
 
-    static class ColorPair {
-        private MajorColor majorColor;
-        private MinorColor minorColor;
-        
-        public ColorPair(MajorColor major, MinorColor minor)
-        {
-            majorColor = major;
-            minorColor = minor;
-        }
-        public MajorColor getMajor() {
-            return majorColor;
-        }
-        public MinorColor getMinor() {
-            return minorColor;
-        }
-        String ToString() {
-            String colorPairStr = MajorColorNames[majorColor.getIndex()];
-            colorPairStr += " ";
-            colorPairStr += MinorColorNames[minorColor.getIndex()];
-            return colorPairStr;
+	static int getPairNumberFromColor(MajorColor major, MinorColor minor) {
```

Comment:
This is production code. the test code (in this case calling of test functions) and the production code are in the same main file.
Code organization may be improved by keeping production code separately from test code. Main also is test code here because it calls test code.

The production code movement to another file would keep it organized.